src/main/java/io/wisetime/connector/jira/database/JiraDb.java

```diff
@@ -36,31 +47,125 @@ public void asTransaction(final Runnable runnable) {
   }
 
   public boolean canUseDatabase() {
-    // TODO
-    return true;
+    return hasCorrectSchema(
+        ImmutableList.of(
+            Pair.of(
+                "jiraissue",
+                ImmutableSet.of("id", "issuenum", "summary", "timespent", "project")
+            ),
+            Pair.of(
+                "project",
+                ImmutableSet.of("id", "pkey")
+            ),
+            Pair.of(
+                "cwd_user",
+                ImmutableSet.of("user_name", "lower_email_address")
+            ),
+            Pair.of(
+                "worklog",
+                ImmutableSet.of("id", "issueid", "author", "timeworked", "created", "worklogbody")
+            ),
+            Pair.of(
+                "sequence_value_item",
+                ImmutableSet.of("seq_id", "seq_name")
+            )
+        )
+    );
+  }
+
+  private boolean hasCorrectSchema(List<Pair<String, Set<String>>> tableAndColumnPairs) {
+    return tableAndColumnPairs.stream()
+        .allMatch(tableColumnPair -> hasCorrectSchema(tableColumnPair.getKey(), tableColumnPair.getValue()));
+  }
+
+  private boolean hasCorrectSchema(String tableName, Set<String> columnNames) {
+    boolean hasTable = query.databaseInspection()
+        .selectFromMetaData(meta -> meta.getTables(null, null, null, null))
+        .listResult(rs -> rs.getString("TABLE_NAME"))
+        .stream()
+        .anyMatch(table -> table.equalsIgnoreCase(tableName));
+
+    boolean hasAllColumns = ImmutableSet.copyOf(
+        query.databaseInspection()
```

Comment:
How expensive is database inspection? Is that something that we want to do regularly whenever `JiraConnector#isConnectorHealthy` is called?

Reply:
Actual query takes place on accessing `DatabaseMetaData` which the DB driver provides. On my observation for MySQL, it doesn't look to be expensive though.